File: src/main/java/teste_tecnico_analista_pleno/domain/Transfer.java
package teste_tecnico_analista_pleno.domain;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor

@Entity
public class Transfer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String originAccount;
    private String destinationAccount;
    private BigDecimal amount;
    private BigDecimal fee;
    private LocalDateTime transferDate;
    private LocalDateTime appointmentDate;

}

File: front_teste_tecnico_analista_pleno/src/types/transfer.ts

export interface TransferRequest {
  originAccount: string;
  destinationAccount: string;
  amount: number;
  transferDate: string;
}

export interface TransferResponse {
  id: number;
  originAccount: string;
  destinationAccount: string;
  amount: number;
  fee: number;
  transferDate: string;
  appointmentDate: string;
}